File: src/types/mcp.ts
/// per-app 启用开关：key 为应用标识符，value 为是否启用
export type McpApps = Record<string, boolean>;

export interface McpServer {
    id: string;
    name: string;
    command?: string;
    args?: string[];
    url?: string;
    env?: Record<string, string>;
    enabled: boolean;
    transport: 'stdio' | 'http' | 'sse';
    source: 'global' | 'project';
    /// per-app 启用开关；undefined 或空对象表示旧数据，视为全部应用启用
    apps?: McpApps;
}

/// 应用标识符常量
export const APP_KEYS = ['claude_code', 'cursor', 'windsurf', 'vscode', 'other'] as const;
export type AppKey = (typeof APP_KEYS)[number];

/// 应用显示标签
export const APP_LABELS: Record<AppKey, string> = {
    claude_code: 'Claude Code',
    cursor: 'Cursor',
    windsurf: 'Windsurf',
    vscode: 'VS Code',
    other: '其他',
};

File: src-tauri/src/models/mcp.rs
#![allow(dead_code)]
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

/// per-app 启用开关：key 为应用标识符（如 "claude_code"），value 为是否启用
pub type McpApps = HashMap<String, bool>;

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum TransportType {
    Stdio,
    Http,
    Sse,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum McpSource {
    Global,
    Project,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct McpServer {
    pub id: String,
    pub name: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub command: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub args: Option<Vec<String>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub url: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub env: Option<HashMap<String, String>>,
    pub enabled: bool,
    pub transport: TransportType,
    pub source: McpSource,
    /// per-app 启用开关；空 map 表示旧数据，视为全部应用启用
    #[serde(default)]
    pub apps: McpApps,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[allow(non_snake_case)]
pub struct McpConfig {
    pub mcpServers: HashMap<String, ServerConfig>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ServerConfig {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub command: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub args: Option<Vec<String>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub url: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub env: Option<HashMap<String, String>>,
    /// per-app 开关，持久化到配置文件；空 map 不序列化（向后兼容）
    #[serde(default, skip_serializing_if = "HashMap::is_empty")]
    pub apps: McpApps,
}
